File: Dummy/hawkshaw/library/preferences/ExtKt.java
package com.hawkshaw.library.preferences;

import Y1.K;
import l3.C0709a;

public final class ExtKt {
    public static final C0709a booleanPreference(String str, boolean z4) {
        K.n(str, "key");
        return new PreferenceProperty(str, Boolean.valueOf(z4), a.f5057L, b.f5058L);
    }

    public static /* synthetic */ C0709a booleanPreference$default(String str, boolean z4, int i5, Object obj) {
        if ((i5 & 2) != 0) {
            z4 = false;
        }
        return booleanPreference(str, z4);
    }

    public static final C0709a floatPreference(String str, float f5) {
        K.n(str, "key");
        return new PreferenceProperty(str, Float.valueOf(f5), c.f5059L, d.f5060L);
    }

    public static /* synthetic */ C0709a floatPreference$default(String str, float f5, int i5, Object obj) {
        if ((i5 & 2) != 0) {
            f5 = 0.0f;
        }
        return floatPreference(str, f5);
    }

    public static final C0709a intPreference(String str, int i5) {
        K.n(str, "key");
        return new PreferenceProperty(str, Integer.valueOf(i5), e.f5061L, f.f5062L);
    }

    public static /* synthetic */ C0709a intPreference$default(String str, int i5, int i6, Object obj) {
        if ((i6 & 2) != 0) {
            i5 = 0;
        }
        return intPreference(str, i5);
    }

    public static final C0709a longPreference(String str, long j5) {
        K.n(str, "key");
        return new PreferenceProperty(str, Long.valueOf(j5), g.f5063L, h.f5064L);
    }

    public static /* synthetic */ C0709a longPreference$default(String str, long j5, int i5, Object obj) {
        if ((i5 & 2) != 0) {
            j5 = 0;
        }
        return longPreference(str, j5);
    }

    public static final C0709a stringPreference(String str, String str2) {
        K.n(str, "key");
        return new PreferenceProperty(str, str2, i.f5065L, j.f5066L);
    }

    public static /* synthetic */ C0709a stringPreference$default(String str, String str2, int i5, Object obj) {
        if ((i5 & 2) != 0) {
            str2 = null;
        }
        return stringPreference(str, str2);
    }
}

File: app/src/main/java/com/hawkshaw/library/preferences/ExtKt.kt
package com.hawkshaw.library.preferences

import android.content.SharedPreferences
import kotlin.properties.ReadWriteProperty
import android.util.Log

/**
 * Extension functions for SharedPreferences property delegates
 */
object ExtKt {
    private const val TAG = "PreferencesExtKt"

    /**
     * Creates a boolean preference property with the given key and default value
     */
    @JvmStatic
    @JvmOverloads
    fun booleanPreference(key: String, defaultValue: Boolean = false): ReadWriteProperty<Any, Boolean?> {
        Log.d(TAG, "[DEBUG] booleanPreference delegate created for Key: \"$key\", DefaultValue: $defaultValue")
        return PreferenceProperty(
            key,
            defaultValue,
            { prefs: SharedPreferences, k: String, default: Boolean? ->
                Log.d(TAG, "[DEBUG] booleanPreference GET for Key: \"$k\", Default supplied: $default")
                val value = if (default != null) prefs.getBoolean(k, default) else false
                Log.d(TAG, "[DEBUG] booleanPreference GET for Key: \"$k\", Retrieved value: $value")
                value
            },
            { editor: SharedPreferences.Editor, k: String, value: Boolean? ->
                Log.d(TAG, "[DEBUG] booleanPreference SET for Key: \"$k\", Value to set: $value")
                if (value != null) {
                    editor.putBoolean(k, value)
                    Log.d(TAG, "[DEBUG] booleanPreference SET for Key: \"$k\", Value $value WILL BE written.")
                } else {
                    editor.remove(k)
                    Log.d(TAG, "[DEBUG] booleanPreference SET for Key: \"$k\", Value was null, key WILL BE removed.")
                }
                editor // Ensure editor is returned
            }
        )
    }

    /**
     * Creates a float preference property with the given key and default value
     */
    @JvmStatic
    @JvmOverloads
    fun floatPreference(key: String, defaultValue: Float = 0f): ReadWriteProperty<Any, Float?> {
        Log.d(TAG, "[DEBUG] floatPreference delegate created for Key: \"$key\", DefaultValue: $defaultValue")
        return PreferenceProperty(
            key,
            defaultValue,
            { prefs: SharedPreferences, k: String, default: Float? ->
                Log.d(TAG, "[DEBUG] floatPreference GET for Key: \"$k\", Default supplied: $default")
                val value = if (default != null) prefs.getFloat(k, default) else 0f
                Log.d(TAG, "[DEBUG] floatPreference GET for Key: \"$k\", Retrieved value: $value")
                value
            },
            { editor: SharedPreferences.Editor, k: String, value: Float? ->
                Log.d(TAG, "[DEBUG] floatPreference SET for Key: \"$k\", Value to set: $value")
                if (value != null) {
                    editor.putFloat(k, value)
                    Log.d(TAG, "[DEBUG] floatPreference SET for Key: \"$k\", Value $value WILL BE written.")
                } else {
                    editor.remove(k)
                    Log.d(TAG, "[DEBUG] floatPreference SET for Key: \"$k\", Value was null, key WILL BE removed.")
                }
                editor // Ensure editor is returned
            }
        )
    }

    /**
     * Creates an integer preference property with the given key and default value
     */
    @JvmStatic
    @JvmOverloads
    fun intPreference(key: String, defaultValue: Int = 0): ReadWriteProperty<Any, Int?> {
        Log.d(TAG, "[DEBUG] intPreference delegate created for Key: \"$key\", DefaultValue: $defaultValue")
        return PreferenceProperty(
            key,
            defaultValue,
            { prefs: SharedPreferences, k: String, default: Int? ->
                Log.d(TAG, "[DEBUG] intPreference GET for Key: \"$k\", Default supplied: $default")
                val value = if (default != null) prefs.getInt(k, default) else 0
                Log.d(TAG, "[DEBUG] intPreference GET for Key: \"$k\", Retrieved value: $value")
                value
            },
            { editor: SharedPreferences.Editor, k: String, value: Int? ->
                Log.d(TAG, "[DEBUG] intPreference SET for Key: \"$key\", Value to set: $value") // Corrected key variable
                if (value != null) {
                    editor.putInt(k, value)
                    Log.d(TAG, "[DEBUG] intPreference SET for Key: \"$k\", Value $value WILL BE written.")
                } else {
                    editor.remove(k)
                    Log.d(TAG, "[DEBUG] intPreference SET for Key: \"$k\", Value was null, key WILL BE removed.")
                }
                editor // Ensure editor is returned
            }
        )
    }

    /**
     * Creates a long preference property with the given key and default value
     */
    @JvmStatic
    @JvmOverloads
    fun longPreference(key: String, defaultValue: Long = 0L): ReadWriteProperty<Any, Long?> {
        Log.d(TAG, "[DEBUG] longPreference delegate created for Key: \"$key\", DefaultValue: $defaultValue")
        return PreferenceProperty(
            key,
            defaultValue,
            { prefs: SharedPreferences, k: String, default: Long? ->
                Log.d(TAG, "[DEBUG] longPreference GET for Key: \"$k\", Default supplied: $default")
                val value = if (default != null) prefs.getLong(k, default) else 0L
                Log.d(TAG, "[DEBUG] longPreference GET for Key: \"$k\", Retrieved value: $value")
                value
            },
            { editor: SharedPreferences.Editor, k: String, value: Long? ->
                Log.d(TAG, "[DEBUG] longPreference SET for Key: \"$key\", Value to set: $value")  // Corrected key variable
                if (value != null) {
                    editor.putLong(k, value)
                    Log.d(TAG, "[DEBUG] longPreference SET for Key: \"$key\", Value $value WILL BE written.")
                } else {
                    editor.remove(k)
                    Log.d(TAG, "[DEBUG] longPreference SET for Key: \"$k\", Value was null, key WILL BE removed.")
                }
                editor // Ensure editor is returned
            }
        )
    }

    /**
     * Creates a string preference property with the given key and default value
     */
    @JvmStatic
    @JvmOverloads
    fun stringPreference(key: String, defaultValue: String? = null): ReadWriteProperty<Any, String?> {
        Log.d(TAG, "[DEBUG] stringPreference delegate created for Key: \"$key\", DefaultValue: \"$defaultValue\"")
        return PreferenceProperty(
            key,
            defaultValue,
            { prefs: SharedPreferences, k: String, default: String? ->
                Log.d(TAG, "[DEBUG] stringPreference GET for Key: \"$k\", Default supplied: \"$default\"")
                val value = prefs.getString(k, default)
                Log.d(TAG, "[DEBUG] stringPreference GET for Key: \"$k\", Retrieved value: \"$value\"")
                value
            },
            { editor: SharedPreferences.Editor, k: String, value: String? ->
                Log.d(TAG, "[DEBUG] stringPreference SET for Key: \"$key\", Value to set: \"$value\"") // Corrected key variable
                if (value != null) {
                    editor.putString(k, value)
                    Log.d(TAG, "[DEBUG] stringPreference SET for Key: \"$key\", Value \"$value\" WILL BE written.")
                } else {
                    editor.remove(k)
                    Log.d(TAG, "[DEBUG] stringPreference SET for Key: \"$key\", Value was null, key WILL BE removed.")
                }
                editor // Ensure editor is returned
            }
        )
    }
}
